plugins/chat/lib/chat/channel_fetcher.rb

```diff
@@ -174,16 +173,46 @@ def self.preload_custom_fields_for(channels)
       )
     end
 
-    def self.secured_direct_message_channels(user_id, memberships, guardian)
-      query = Chat::Channel.includes(chatable: [{ direct_message_users: :user }, :users])
+    def self.secured_direct_message_channels(user_id, guardian)
```

Comment:
I tried to write tests for this, but the whole chat_channel_fetche_spec is quite complicated to navigate, I will do a follow-up to rewrite all of this. Between the existing specs and my system specs it's already more or less covered.

I have few notes here:

- `def self.secured_direct_message_channels(user_id, memberships, guardian)` the previous signature was using `memberships` but this was used nowhere

- `generate_allowed_channel_ids_sql(guardian)` is always including `category_channels` even when we fetch only `direct_message_channels`

- In the end I didn't use it because I wanted a better query to look for direct message channels of a user but who also have one of the given user_ids